rotkehlchen/exchanges/htx.py

```diff
@@ -208,4 +362,94 @@ def query_online_trade_history(
+        to filter the data that we need.
+
+        https://www.htx.com/en-us/opend/newApiPages/?id=7ec4de21-7773-11ed-9966-0242ac110003
+        """
+        new_trades: list[Trade] = []
+        upper_ts, lower_ts = end_ts, Timestamp(end_ts - DAY_IN_SECONDS * 2)
+        earliest_query_start_ts = Timestamp(ts_now() - DAY_IN_SECONDS * 120)
+
+        if end_ts <= earliest_query_start_ts:  # 0... start_ts ... end_ts ... earliest_query_start
+            return [], (start_ts, end_ts)  # entire query out of range. Bail
+
+        if start_ts <= earliest_query_start_ts:  # 0 ... start_ts ...  earliest_query_start ... end_ts  # noqa: E501
+            start_ts = Timestamp(earliest_query_start_ts + 60 * 5)  # margin of 5 minutes to avoid possible errors in case of delay  # noqa: E501
+
+        while True:
+            raw_data = self._query(
+                absolute_path='/v1/order/matchresults',
+                options={
+                    'start-time': str(ts_sec_to_ms(lower_ts)),
+                    'end-time': str(ts_sec_to_ms(upper_ts)),
+                    'size': PAGINATION_LIMIT,
+                },
+            )
+            for raw_trade in raw_data:
+                try:
+                    symbol, fee_currency = raw_trade['symbol'], raw_trade['fee-currency']
+                    trade_type = TradeType.deserialize(raw_trade['type'].split('-')[0])
+                    # On the API page, it's mentioned transaction fee of buy order is based on
+                    # base currency, transaction fee of sell order is based on quote currency
+                    if trade_type == TradeType.BUY:
+                        base_asset, quote_asset = asset_from_htx(fee_currency), asset_from_htx(
+                            remove_fee_currency(symbol=symbol, fee_currency=fee_currency),
+                        )
+                        fee_asset = base_asset
+                    elif trade_type == TradeType.SELL:
+                        base_asset, quote_asset = asset_from_htx(
+                            remove_fee_currency(symbol=symbol, fee_currency=fee_currency),
+                        ), asset_from_htx(fee_currency)
+                        fee_asset = quote_asset
+                    else:
+                        raise UnprocessableTradePair(raw_trade['symbol'])
+                except UnknownAsset:
+                    log.error(f'Could not read assets from HTX trade {raw_trade}')
+                    continue
+                except UnprocessableTradePair:
+                    log.error(f'Could not process base and quote assets from HTX trade {raw_trade}')  # noqa: E501
+                    continue
+                except KeyError as e:
+                    log.error(f'Key `{e!s}` missing from HTX trade {raw_trade}')
+                    continue
+
+                try:
+                    trade = Trade(
+                        timestamp=deserialize_timestamp_from_intms(int(raw_trade['created-at'])),
```

Comment:
I believe the int conversion can raise valueERror. Use `deserialize_int_from_str`